rpc-client-api/src/filter.rs

```diff
@@ -96,96 +78,100 @@ impl RpcFilterType {
 pub enum RpcFilterError {
     #[error("encoded binary data should be less than 129 bytes")]
     DataTooLarge,
-    #[deprecated(
-        since = "1.8.1",
-        note = "Error for MemcmpEncodedBytes::Binary which is deprecated"
-    )]
-    #[error("encoded binary (base 58) data should be less than 129 bytes")]
-    Base58DataTooLarge,
-    #[deprecated(
-        since = "1.8.1",
-        note = "Error for MemcmpEncodedBytes::Binary which is deprecated"
-    )]
-    #[error("bs58 decode error")]
-    DecodeError(bs58::decode::Error),
     #[error("base58 decode error")]
     Base58DecodeError(#[from] bs58::decode::Error),
     #[error("base64 decode error")]
     Base64DecodeError(#[from] base64::DecodeError),
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub enum MemcmpEncoding {
-    Binary,
-}
-
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase", untagged)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize)]
+#[serde(rename_all = "camelCase", tag = "encoding", content = "bytes")]
 pub enum MemcmpEncodedBytes {
-    #[deprecated(
-        since = "1.8.1",
-        note = "Please use MemcmpEncodedBytes::Base58 instead"
-    )]
-    Binary(String),
     Base58(String),
     Base64(String),
     Bytes(Vec<u8>),
 }
 
+impl<'de> Deserialize<'de> for MemcmpEncodedBytes {
```

Comment:
This part is unfortunate. It only exists in order to support RPC calls with a filter that does not include an `encoding` field. (Serde's adjacent tagging cannot handle a missing tag.) If we were okay suddenly requiring `encoding` despite that not being previously documented, this could all be removed. Wdyt?

Reply:
do pre-existing tests cover this logic?

Reply:
Only sort of. I can add a test that shows how serialization/deserialization changes (or doesn't).

Reply:
Okay, test added in https://github.com/anza-xyz/agave/pull/2050/commits/632035ffc1d0fcb9282d89d6f361396fd2d1175d
Updated in https://github.com/anza-xyz/agave/pull/2050/commits/d25d169911f171cab2ced84201dbd4c311706bf9

There is one small change in behavior; with adjacently tagged serialization, the raw-bytes filter serializes the `encoding` field as `"bytes"` instead of `null`. But I added a test case in the original test to show that such a json filter would already be deserialized properly before this PR.